src/shared/tree_container.h

```diff
@@ -133,16 +133,20 @@ class TreeContainer
     const fs::path &name, const T &data,
     TreeFlags flags = 0, bool overwrite = true)
   {
-    try
-    {
-      return addNode(
-        m_TreeMeta->tree.get(), name, name.begin(),
-        data, overwrite, flags, allocator());
-    }
-    catch (const bi::bad_alloc&)
-    {
+    for (;;) {
+      DecomposablePath dp(name.string());
```

Comment:
Could this go out of the for loop? I mean I don't think it really matters as if I understand correctly the loop only happens if there is a reassign

Reply:
No, it's taken by reference and modified to walk the path components.

Reply:
Ah ok, thanks